st2common/tests/unit/test_db.py

```diff
@@ -36,6 +36,17 @@
 from st2common.persistence.rule import Rule
 from st2common.persistence.trigger import TriggerType, Trigger, TriggerInstance
 from st2tests import DbTestCase
+from st2tests.mocks import liveaction as mock_liveaction
+
+
+# Previous test disrupts state for tests in this module. We reload the import
+# to avoid this. This try/except block is for python 3 support.
+try:
+    from imp import reload
+except:
+    from importlib import reload
+
+reload(mock_liveaction)
```

Comment:
Yeah, that's nasty :/

Every tests which use that mock should have ``.wait_all()`` call in ``tearDown()`` to  ensure mock liveaction has processed all the pending work.